### ISSUE TITLE
Cloudvars unable to be created after account migration.

### ISSUE THREAD

**Issue by Gyztor** [Yellow-Dog-Man/Resonite-Issues#3101]
### Describe the bug?

I have recently migrated an account from Neos and have attempted to merge a previous resonite account (which so far seems to have been successful) other than the fact that I cannot under any circumstances create NEW cloudvars. Old ones that were migrated mostly work fine but any new ones (so far tested with only string based Cloudvars) don't actually show up as useable or readable in game despite the resonite bot fully showing it exists. Log files show nothing other than attempts to sync cloudvar.

Something that may be related for this problem is since migration the "My Account" has show that it sent a contact request and ignoring or accepting does nothing to change this and every boot it shows a contact request has been sent again.
![Image](https://github.com/user-attachments/assets/e9aabf3a-590f-4686-90a0-b30aa41a8035)

### To Reproduce

1. Migrate an account from resonite -> resonite where previous resonite account had Cloudvars.
2. Attempt to create a new string Cloudvar after full migration.
3. Cloudvar never shows up in playspace or local space and is unusable.

### Expected behavior

Cloudvars work and appear after creation.

### Screenshots

![Image](https://github.com/user-attachments/assets/7b66d9a9-6f27-4d11-a1c5-ff9bf82a65f6)
![Image](https://github.com/user-attachments/assets/1f733512-7dab-4d64-b121-3f18d12ad274)

### Resonite Version Number

Beta 2024.10.8.1349

### What Platforms does this occur on?

Windows, Linux

### What headset if any do you use?

Quest 3 via Virtual Desktop

### Log Files

[DESKTOP-4P84S13 - 2024.10.8.1349 - 2024-10-20 21_06_52.log](https://github.com/user-attachments/files/17453297/DESKTOP-4P84S13.-.2024.10.8.1349.-.2024-10-20.21_06_52.log)

### Reporters

GyztorMizirath

**Comment by Gyztor** [comment]
I attempted to update the perms on some old vars and now those are no longer working

**Comment by Frooxius** [comment]
Can you share the user ID of the account you migrated from?

**Comment by Gyztor** [comment]
Yes. I can get that. As for "No Longer Working" i mean in every way a cloud var is supposed to function they simply do not, with no errors being displayed. When attempting to read them they show the "failed" status with no logs to explain why. the Read Cloudvar component also does not link to the variable at all. I wish i had a more apt description but there really isn't any more than that. it simply does not work and nothing on my side gives any other feedback as to why.

Old Resonite Account ID: U-1UedQOwLdIG

**Comment by Frooxius** [comment]
What happened when you tried to update the permissions? Did the command respond? When you get the definition, does it show updated perms?

**Comment by Gyztor** [comment]
Yes all of the resonite bot commands act and work correctly and show updated definitions and permissions but they don't read nor can be updated through protoflux (if write is set to unsafe)

**Comment by shiftyscales** [comment]
Additional context has been provided- seeking further evaluation from @Frooxius.

**Comment by shiftyscales** [comment]
Sounds like there may be another instance of this issue in #3133 - that report pertains to group variables however- so I don't know with certainty if it has the exact same issue- but it sounds likely to at least share an underlying issue.

**Comment by Frooxius** [comment]
I can't replicate this on my end. I've created a new cloud variable on migrated account, read from it and it worked fine.

How do you know that this is specifically affecting migrated accounts? Does this work for accounts that aren't migrated?

Is anyone else with migrated account experiencing the same problem?

**Comment by Gyztor** [comment]
I'm not entirely sure exactly what to say as I have no further information realistically on my end as it's something not on my client. I have asked other people who have migrated and they don't have any problems they can easily point out or don't use cloudvars period to previously having stuff break to never be fixed (don't have further context on that sadly). I have tried every physical combination and nothing works all cloudvars never show the "Linked to Cloud" switch or fail to read. Cloudvars like cloud colors work perfectly fine. Only ones created by myself are unreadable or properly settable by myself. Other people can read the values in the cloudvars though which is interesting. This also seems to 100% relate to my inability to save to groups and the fact that I constantly have a contact request from myself and am unable to accept it. My best guess right now is this is something to do with how cloudvars fundamentally work using a contact of ones self to read them, but that is only a guess as I would literally have no way to know how it works as I have done everything physically possible on my side to get them to work. I'm not sure if it was a bug that came about from migrating and not having enough storage on the new account when migrating, Old Res account -> New Res account. I wish I had a better response but I honestly don't have anything better as it's not a client issue as trying on multiple computers does nothing as well.

**Comment by Gyztor** [comment]
any updates this has been taking unreasonably long and is delaying me too much now.

**Comment by shiftyscales** [comment]
Have you identified any other users that have this issue, @Gyztor? As you'd mentioned above- it seems other users, including migrated ones do not experience this issue, nor does it seem to be able to be replicated trivially per Frooxius's attempt above.

While I'm sorry this issue has been affecting you so prominently I do not know when this could be prioritized since it does not seem to be a widespread issue, nor is it easily replicable.

It would just require time to look into it- but that same time can also be spent on issues that affect a wider number of users, or in more serious ways.

I hope you understand. If you find any additional steps to replicate that behaviour- or other users that are affected by this issue, please let us know- until this can be looked into further, I am sorry for the inconvenience.

**Comment by Gyztor** [comment]
This issue is a circumstance issue. I do believe it is something with just my migration and the issues that came along with it vis storage limitations causing an error during migrating corrupting my ability to make cloud vars and the most likely easily fixable issue being down to the fact my account is not "friends" or "contacts" with itself as mentioned in the extra information i found that could be related above. I believe it's something that could easily be fixed by just forcing the system to accept the broken contact request with myself, than seeing if that resolved it and just doing trial by error. But this issue is just not getting that in the slightest and really it's infuriating to the point i refuse to make anything else until this is resolved because it literally breaks any and all systems i have and attempt to make. The problem couldn't be recreated, okay well the next thing is to poke my specific problem with a stick until it works, which i would love to do but don't have the access to do so. I am sorry if i seem frustrated at all or angry, but i am not the only one who has looked at this issue and also thought the same thing of "why not just solve this in a case by case basis due to its rarity and just get this closed and tally down how much ot happens". This has been objectively handled terribly especially for something that could be fixed in about 5 minutes if you know how a database works.

**Comment by shiftyscales** [comment]
As above- I understand you are frustrated- but please keep your language objective and neutral per our reporting requirements and code of conduct, @Gyztor.

Insults are not going to see your issue resolved any quicker. We are a small team, and we can't prioritize everything at once. Please be reasonable.

@ProbablePrime might possibly be able to look into this issue at some point as well if Frooxius is / has been unable to.

**Comment by Gyztor** [comment]
Issue has gotten a bit worse as of late. now when i set a cloud var in my group or using flux it reverts to a null state after a cache clear or something similar (aka usually after 1 play session). This happens regardless of how I change it. Cloudvars created by my user are still unaccessible due to my user not being contacts with itself.

**Comment by Gyztor** [comment]
Have now noticed that new variables that use the Variable_owner and other similar things now no longer read when I write a variable for myself. Still seems related to my user account not being Contacts with itself.
